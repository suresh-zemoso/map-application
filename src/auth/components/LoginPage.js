import React, { useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import LoginBlock from './LoginBlock';
import signout from '../../utils/signout';
import PropTypes from 'prop-types';



const useStyles = makeStyles({
    root: {
        display: 'flex',
        justifyContent: 'center',
        width: '100%',
        height: '100%',
        background: 'white'
    },
});

const defaultProps = {

}

const propTypes = {


}

const LoginPage = (props) => {

    const classes = useStyles(props);

    useEffect(() => {
        signout()
    }, [])
    return (
        <div className={classes.root}>
            <LoginBlock />
        </div>
    );
};


LoginPage.propTypes = propTypes;
LoginPage.defaultProps = defaultProps;

export default LoginPage;