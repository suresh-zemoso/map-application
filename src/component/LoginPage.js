import React, { useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import LoginBlock from './LoginBlock';
import signout from '../utility/signout';


const useStyles = makeStyles({
    root: {
        display: 'flex',
        justifyContent: 'center',
        width: '100%',
        height: '100%',
        background: 'white'
    },
});


const LoginPage = (props) => {

    const classes = useStyles(props);

    useEffect(() => {
        signout()
    }, [])
    return (
        <div className={classes.root}>
            <LoginBlock />
        </div>
    );
};


export default LoginPage;