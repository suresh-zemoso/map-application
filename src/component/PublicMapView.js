import React from 'react';
import BaseMapView from './BaseMapView';

export default function PublicMapView() {
    return (
        <BaseMapView mapAttribute={{ lng: 5, lat: 34, zoom: 2 }} mapMode="streets-v11" />
    )
}
