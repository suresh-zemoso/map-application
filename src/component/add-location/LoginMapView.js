import React, { useState } from 'react';
import BaseMapView from '../public-map/BaseMapView';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { useSelector } from "react-redux";
import { locationService } from '../../services/locationService';





const useStyles = makeStyles({
    sidebarStyle: {
        position: 'absolute',
        backgroundColor: '#404040',
        color: '#ffffff',
        zIndex: '1!important',
        padding: '6px',
        fontWeight: 'bold'
    }
});

export default function LoginMapView(props) {

    const [mapAttribute, setMapAttribute] = useState({ lng: 73.413827, lat: 25.676973, zoom: 2 });
    const [open, setOpen] = React.useState(false);

    const accessToken = useSelector(state => state.authentication.accessToken);
    const classes = useStyles(props);

    const handleClickOpen = () => {
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };

    function addMapOnclick(map) {
        map.on('click', (e) => {
            setMapAttribute({
                lng: e.lngLat.lng.toFixed(6),
                lat: e.lngLat.lat.toFixed(6),
                zoom: map.getZoom().toFixed(2)
            });
        });
    }
    function addLocation(mapAttribute) {
        const data = {
            lng: mapAttribute.lng,
            lat: mapAttribute.lat
        }
        locationService.addLocation(data, accessToken)
            .then(response => {

            })
            .catch(error => {
                console.log(error);
            })
        handleClose();
    }
    return (
        <>
            <BaseMapView
                handlers={[addMapOnclick]}
                mapAttribute={{
                    lng: mapAttribute.lng,
                    lat: mapAttribute.lat,
                    zoom: mapAttribute.zoom
                }}
                mapMode="streets-v11"
                buttons={[{ buttonText: "Add Location", onClick: () => handleClickOpen() }]}
            >
                <div className={classes.sidebarStyle}>
                    <div>Longitude: {mapAttribute.lng} | Latitude: {mapAttribute.lat}</div>
                </div>
            </BaseMapView >
            <Dialog
                open={open}
                onClose={handleClose}
                aria-labelledby="alert-dialog-title"
                aria-describedby="alert-dialog-description"
            >
                <DialogTitle id="alert-dialog-title">{"Save Location"}</DialogTitle>
                <DialogContent>
                    <DialogContentText id="alert-dialog-description">
                        Do you want to save location with Longitude {mapAttribute.lng} and Latitude {mapAttribute.lat} ?
          </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Button onClick={handleClose} color="primary">
                        Cancel
          </Button>
                    <Button onClick={() => addLocation(mapAttribute)} color="primary" autoFocus>
                        Confirm
          </Button>
                </DialogActions>
            </Dialog>
        </>
    )
}
