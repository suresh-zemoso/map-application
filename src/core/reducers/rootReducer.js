import { combineReducers } from 'redux'
import { authentication } from '../../auth/reducers/loginReducer';
import { locationState } from '../../location/reducers/locationReducer';
import storage from 'redux-persist/lib/storage';
import { persistReducer } from 'redux-persist';



const persistConfig = {
    key: 'root',
    storage,
    blacklist: ['authentication', 'locationState']
}
const authPersistConfig = {
    key: 'authentication',
    storage,
    blacklist: ['error']
}
const locationPersistConfig = {
    key: 'locationState',
    storage,
    blacklist: ['fetchError', 'loading']
}

const rootReducer = combineReducers({
    authentication: persistReducer(authPersistConfig, authentication),
    locationState: persistReducer(locationPersistConfig, locationState)
})

export default persistReducer(persistConfig, rootReducer)
