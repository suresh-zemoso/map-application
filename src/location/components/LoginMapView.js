import React, { useState, useRef } from 'react';
import BaseMapView from '../../location/components/BaseMapView';
import { makeStyles } from '@material-ui/core/styles';
import { useSelector, useDispatch } from "react-redux";
import Snackbar from '@material-ui/core/Snackbar';
import Alert from '@material-ui/lab/Alert';
import { addLocation } from '../../location/actions/locationAction';
import mapboxgl from 'mapbox-gl';
import PropTypes from 'prop-types';




const useStyles = makeStyles({
    sidebarStyle: {
        position: 'absolute',
        backgroundColor: '#404040',
        color: '#ffffff',
        zIndex: '1!important',
        padding: '6px',
        fontWeight: 'bold'
    },
    marker: {
        background: `url(${require("./mapbox-icon.svg")})`,
        backgroundRepeat: 'no-repeat',
        backgroundSize: '45px 45px',
        width: '50px',
        height: '50px',
        borderRadius: '50%',
        cursor: 'pointer'
    },
    snackbar: {
        left: '63% !important'
    }
});


const defaultProps = {

}

const propTypes = {


}
const LoginMapView = (props) => {

    const [mapAttribute, setMapAttribute] = useState({ lng: 73.413827, lat: 25.676973, zoom: 2 });
    const [isSnackbarOpen, setSnackbarOpen] = useState(false);
    const [messageInfo, setMessageInfo] = useState(undefined)
    const queueRef = useRef([])

    const accessToken = useSelector(state => state.authentication.accessToken);
    const saveError = useSelector(state => state.locationState.saveError);

    const dispatch = useDispatch();

    const classes = useStyles(props);

    function formatMessage(lat, lng) {
        return `Location with Latitude ${lat} and Longitude ${lng} Added`
    }

    function processQueue() {
        if (queueRef.current.length > 0) {
            setMessageInfo(queueRef.current.shift());
            setSnackbarOpen(true);
        }
    };

    function openSnackbar(message) {
        queueRef.current.push({
            message,
            key: new Date().getTime(),
        });

        if (isSnackbarOpen) {
            setSnackbarOpen(false);
        } else {
            processQueue();
        }
    };

    function handleClose(event, reason) {
        if (reason === 'clickaway') {
            return;
        }
        setSnackbarOpen(false);
    };

    function handleExited() {
        processQueue();
    };

    function mousemoveListener(map, e) {
        setMapAttribute({
            lng: e.lngLat.lng.toFixed(6),
            lat: e.lngLat.lat.toFixed(6),
            zoom: map.getZoom().toFixed(2)
        });
    }

    function onClickListener(map, e) {
        // create a HTML element for each feature
        var el = document.createElement('div');
        el.className = classes.marker;
        console.debug("Lat lang value", e.lngLat.lat, e.lngLat.lng)
        // make a marker for each feature and add to the map
        const location = {
            lng: e.lngLat.lng.toFixed(6),
            lat: e.lngLat.lat.toFixed(6)
        }

        dispatch(addLocation(location, accessToken))
            .then(data => {
                openSnackbar(formatMessage(location.lat, location.lng));
                new mapboxgl.Marker(el)
                    .setLngLat([e.lngLat.lng, e.lngLat.lat])
                    .addTo(map);
            },
                error => {
                    openSnackbar("Error in adding location");
                    console.log(error)
                })
    }

    const handlers = [{
        addEventListener: (map) => map.on('mousemove', (e) => mousemoveListener(map, e)),
        removeEventListener: (map) => map.off('mousemove', (e) => mousemoveListener(map, e))
    },
    {
        addEventListener: (map) => map.on('click', (e) => onClickListener(map, e)),
        removeEventListener: (map) => map.off('click', (e) => onClickListener(map, e))
    }
    ]


    return (
        <div>
            <BaseMapView
                handlers={handlers}
                mapAttribute={{
                    lng: mapAttribute.lng,
                    lat: mapAttribute.lat,
                    zoom: mapAttribute.zoom
                }}
                mapMode="streets-v11"
            >
                <div className={classes.sidebarStyle}>
                    <div>Longitude: {mapAttribute.lng} | Latitude: {mapAttribute.lat}</div>
                </div>
            </BaseMapView >
            <Snackbar
                key={messageInfo ? messageInfo.key : undefined}
                anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
                open={isSnackbarOpen}
                onExited={handleExited}
                classes={{ anchorOriginTopCenter: classes.snackbar }}
                // message={messageInfo ? messageInfo.message : undefined}
                // autoHideDuration={3000}
                onClose={handleClose} >
                <Alert severity={saveError ? "error" : "success"}>
                    {messageInfo ? messageInfo.message : undefined}
                </Alert>
            </Snackbar>
        </div>
    )
}

LoginMapView.propTypes = propTypes;
LoginMapView.defaultProps = defaultProps;

export default LoginMapView;