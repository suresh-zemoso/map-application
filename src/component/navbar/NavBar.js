import React from "react";
import {
    BrowserRouter as Router,
    Switch,
    Route,
    NavLink
} from "react-router-dom";
import LoginPage from "../login/LoginPage";
import PublicMapView from "../public-map/PublicMapView";
import LoginMapView from "../add-location/LoginMapView";
import LocationList from "../location-list/LocationList";
import { ProtectedRoute } from "../navbar/ProtectedRoute";
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import { useSelector } from "react-redux";
import signout from "../../utility/signout"
import MarkLocation from "../location-view/MarkLocation";



const useStyles = makeStyles({
    button: {
        width: "100%",
        height: "12vh"
    }
});

const routes = [
    {
        path: "/",
        exact: true,
        main: () => <LoginPage />
    },
    {
        path: "/map",
        exact: true,
        main: () => <PublicMapView />
    }
];

const protectedRoutes = [
    {
        path: "/locations/new",
        exact: true,
        main: () => <LoginMapView />
    },
    {
        path: "/locations",
        exact: true,
        main: () => <LocationList />
    },
    {
        path: "/locations/:id",
        exact: true,
        main: () => <MarkLocation />
    }
]

const navBarContent = {
    beforeLogin: [
        {
            to: "/",
            exact: true,
            sidebar: "Login",
        },
        {
            to: "/map",
            exact: true,
            sidebar: "Map",
        }],
    afterLogin: [
        {
            to: "/locations/new",
            exact: true,
            sidebar: "Add Location",
        },
        {
            to: "/locations",
            exact: true,
            sidebar: "View Locations",
        },
        {
            to: "",
            exact: true,
            sidebar: "Logout",
            onClick: () => signout()
        }
    ]
}

export default function NavBar(props) {

    const classes = useStyles(props);
    const isLogin = useSelector(state => state.authentication.isLogin);

    return (
        <Router>
            <div style={{ display: "flex", height: "100%", width: "100%" }}>
                <div
                    style={{
                        padding: "10px",
                        width: "20%",
                        background: "#f0f0f0",
                    }}
                >
                    {!isLogin && navBarContent.beforeLogin.map((route, index) => (
                        <Button component={NavLink}
                            key={index}
                            to={route.to}
                            style={{ textDecoration: "none" }}
                            activeStyle={{
                                background: "#3f51b5",
                                color: "white"
                            }}
                            isActive={(match, location) => {
                                return route.to === location.pathname
                            }} onClick={route.onClick} className={classes.button}>{route.sidebar}</Button>
                    ))}
                    {isLogin &&
                        navBarContent.afterLogin.map((route, index) => (
                            <Button
                                key={index}
                                to={route.to}
                                component={NavLink}
                                style={{ textDecoration: "none" }}
                                activeStyle={{
                                    background: "#3f51b5",
                                    color: "white"
                                }}
                                isActive={(match, location) => {
                                    return route.to === location.pathname
                                }} onClick={route.onClick} className={classes.button}>{route.sidebar}</Button>
                        ))
                    }
                </div>

                <div style={{ width: "80%" }}>
                    <Switch>
                        {routes.map((route, index) => (
                            <Route
                                key={index}
                                path={route.path}
                                exact={route.exact}
                                component={route.main}
                            />
                        ))}
                        {protectedRoutes.map((route, index) => (
                            <ProtectedRoute
                                key={index}
                                path={route.path}
                                exact={route.exact}
                                isLogin={isLogin}
                                component={route.main}
                            />
                        ))}
                    </Switch>
                </div>
            </div>
        </Router>
    );
}
