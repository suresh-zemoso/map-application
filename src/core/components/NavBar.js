import React from "react";
import {
    Switch,
    Route,
    NavLink
} from "react-router-dom";
import LoginPage from "../../auth/components/LoginPage";
import PublicMapView from "../../location/components/PublicMapView";
import LoginMapView from "../../location/components/LoginMapView";
import LocationList from "../../location/components/LocationList";
import { ProtectedRoute } from "../../core/components/ProtectedRoute";
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import { useSelector } from "react-redux";
import signout from "../../utils/signout"
import MarkLocation from "../../location/components/MarkLocation";
import PropTypes from 'prop-types';




const useStyles = makeStyles({
    button: {
        width: "100%",
        height: "12vh"
    }
});

const defaultProps = {

}

const propTypes = {


}


const routes = [
    {
        path: "/",
        exact: true,
        main: () => <LoginPage />
    },
    {
        path: "/map",
        exact: true,
        main: () => <PublicMapView />
    }
];

const protectedRoutes = [
    {
        path: "/locations/new",
        exact: true,
        main: () => <LoginMapView />
    },
    {
        path: "/locations",
        exact: true,
        main: () => <LocationList />
    },
    {
        path: "/locations/:id",
        exact: true,
        main: () => <MarkLocation />
    }
]

const navBarContent = {
    beforeLogin: [
        {
            to: "/",
            exact: true,
            sidebar: "Login",
        },
        {
            to: "/map",
            exact: true,
            sidebar: "Map",
        }],
    afterLogin: [
        {
            to: "/locations/new",
            exact: true,
            sidebar: "Add Location",
        },
        {
            to: "/locations",
            exact: true,
            sidebar: "View Locations",
        },
        {
            to: "",
            exact: true,
            sidebar: "Logout",
            onClick: () => signout()
        }
    ]
}

const NavBar = (props) => {

    const classes = useStyles(props);
    // const state = useSelector(state => state)
    // console.debug("whole redux is", state);
    const isLogin = useSelector(state => state.authentication.isLogin);

    return (
        <div style={{ display: "flex", height: "100%", width: "100%" }}>
            <div
                style={{
                    padding: "10px",
                    width: "20%",
                    background: "#f0f0f0",
                }}
            >
                {!isLogin && navBarContent.beforeLogin.map((route, index) => (
                    <Button component={NavLink}
                        key={index}
                        to={route.to}
                        style={{ textDecoration: "none" }}
                        activeStyle={{
                            background: "#3f51b5",
                            color: "white"
                        }}
                        isActive={(match, location) => {
                            return route.to === location.pathname
                        }} onClick={route.onClick} className={classes.button}>{route.sidebar}</Button>
                ))}
                {isLogin &&
                    navBarContent.afterLogin.map((route, index) => (
                        <Button
                            key={index}
                            to={route.to}
                            component={NavLink}
                            style={{ textDecoration: "none" }}
                            activeStyle={{
                                background: "#3f51b5",
                                color: "white"
                            }}
                            isActive={(match, location) => {
                                return route.to === location.pathname
                            }} onClick={route.onClick} className={classes.button}>{route.sidebar}</Button>
                    ))
                }
            </div>

            <div style={{ width: "80%", overflowY: 'auto' }}>
                <Switch>
                    {routes.map((route, index) => (
                        <Route
                            key={index}
                            path={route.path}
                            exact={route.exact}
                            component={route.main}
                        />
                    ))}
                    {protectedRoutes.map((route, index) => (
                        <ProtectedRoute
                            key={index}
                            path={route.path}
                            exact={route.exact}
                            isLogin={isLogin}
                            component={route.main}
                        />
                    ))}
                </Switch>
            </div>
        </div>
    );
}

NavBar.propTypes = propTypes;
NavBar.defaultProps = defaultProps;

export default NavBar;