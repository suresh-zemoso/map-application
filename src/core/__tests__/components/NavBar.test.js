import React from 'react';
import { mount, shallow } from 'enzyme';
import NavBar from '../../components/NavBar';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import LoginPage from '../../../auth/components/LoginPage';
import * as ReactReduxHooks from '../../../hooks/reduxHooks';
import { mapUtility } from '../../../utils/mapUtility';
import PublicMapView from '../../../location/components/PublicMapView';
import LoginMapView from '../../../location/components/LoginMapView';
import { locationState } from '../../../location/reducers/locationReducer';


jest.mock('mapbox-gl', () => {
    const mockMap = jest.fn(() => ({
        on: jest.fn(),
        off: jest.fn(),
        getSource: jest.fn(),
        getLayer: jest.fn(),
        addControl: jest.fn(),
        addSource: jest.fn(),
        addLayer: jest.fn(),
        addImage: jest.fn(),
        loadImage: jest.fn(),
        hasImage: jest.fn(),
        dragPan: {
            enable: jest.fn(),
            disable: jest.fn()
        },
        scrollZoom: {
            enable: jest.fn(),
            disable: jest.fn()
        },
        doubleClickZoom: {
            enable: jest.fn(),
            disable: jest.fn()
        },
        touchZoomRotate: {
            enable: jest.fn(),
            disable: jest.fn()
        },
        setStyle: jest.fn(),
        queryRenderedFeatures: jest.fn(),
        getCenter: jest.fn(),
        flyTo: jest.fn(),
        isStyleLoaded: jest.fn(),
        getBounds: jest.fn(),
        getZoom: jest.fn(),
        resize: jest.fn(),
        getStyle: jest.fn(() => ({
            layers: []
        })),
        isMoving: jest.fn(),
        isZooming: jest.fn(),
        isRotating: jest.fn(),
        removeControl: jest.fn(),
        remove: jest.fn()
    }));

    const mockPopup = jest.fn(() => ({
        setLngLat: jest.fn(() => mockPopup()),
        setDOMContent: jest.fn(() => mockPopup()),
        remove: jest.fn(() => mockPopup()),
        addTo: jest.fn(() => mockPopup()),
        setMaxWidth: jest.fn(() => mockPopup())
    }));

    const mockMarker = jest.fn(() => ({
        setLngLat: jest.fn(() => mockMarker()),
        remove: jest.fn(() => mockMarker()),
        addTo: jest.fn(() => mockMarker()),
        getElement: jest.fn(() => mockMarker()),
        on: jest.fn(),
        getLngLat: jest.fn()
    }));

    return {
        Map: mockMap,
        NavigationControl: jest.fn(),
        GeolocateControl: jest.fn(() => ({
            on: jest.fn(),
            trigger: jest.fn()
        })),
        Popup: mockPopup,
        LngLat: jest.fn((lat, lng) => ({
            lat, lng,
            wrap: jest.fn(() => ({
                lat, lng
            }))
        })),
        Marker: mockMarker,
        ScaleControl: jest.fn()
    };
});

describe('NavBar', () => {

    let wrapper;
    let store;

    beforeEach(() => {
        /* mocking store */
        store = configureStore([thunk])({
            authentication: { isLogin: false }
        });

        jest
            .spyOn(ReactReduxHooks, "useSelector")
            .mockImplementation(state => store.getState());
    });

    it('should render the NavBar Component correctly', () => {
        wrapper = shallow(<Provider store={store} ><MemoryRouter nitialEntries={['/random']}><NavBar /></MemoryRouter></Provider>);
        expect(wrapper).toMatchSnapshot();
    });

    it('It will not render any page', () => {
        wrapper = mount(<Provider store={store}><MemoryRouter initialEntries={['/random']}><NavBar /></MemoryRouter></Provider >);
        expect(wrapper.find(LoginPage)).toHaveLength(0);
    });

    it('should render login page on / initial entry', () => {
        wrapper = mount(<Provider store={store}><MemoryRouter initialEntries={['/']}><NavBar /></MemoryRouter></Provider>);
        expect(wrapper.find(LoginPage)).toHaveLength(1);
    });
    it('should render public map page on /map initial entry', () => {
        jest.spyOn(mapUtility, "getGeoCoder").mockImplementation(() => { })
        wrapper = mount(<Provider store={store}><MemoryRouter initialEntries={['/map']}><NavBar /></MemoryRouter></Provider>);
        expect(wrapper.find(PublicMapView)).toHaveLength(1);
    });

    it('should render LoginPage when Protected route is accessed and user is not logged in',
        () => {
            wrapper = mount(<Provider store={store}><MemoryRouter initialEntries={['/locations/new']}><NavBar /></MemoryRouter></Provider>);
            expect(wrapper.find(LoginPage)).toHaveLength(1);
        });
    it('should render LoginMapView page when user is logged in and url is /add/location',
        () => {
            store = configureStore([thunk])({
                authentication: { isLogin: true },
                locationState: {
                    loading: true,
                    locations: [],
                    fetchError: false,
                    errorMessage: "Error in Fetching state",
                    saveError: false,
                }
            });
            wrapper = mount(<Provider store={store}><MemoryRouter initialEntries={['/locations/new']}><NavBar /></MemoryRouter></Provider>);
            expect(wrapper.find(LoginMapView)).toHaveLength(1);
        });

});      