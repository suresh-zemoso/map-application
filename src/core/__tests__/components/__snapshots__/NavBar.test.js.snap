// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render the NavBar Component correctly 1`] = `ShallowWrapper {}`;
